### DISCUSSION TITLE
`FocusScope`: the `restoreFocus` prop doesn't restore the focus to the trigger element when one of the children elements have an `autoFocus` prop

### DISCUSSION THREAD

**Discussion by balazsmatepetro** [adobe/react-spectrum#2077] [Q&A]
First of all, I would like to say thank you for building this amazing library, it saved us a lot of time :pray: .

As you can see in the title, I have a question about the `restoreFocus` prop of the `FocusScope` component. When one of the children elements have an `autoFocus` prop that element gets the focus, which is the expected behavior, it's fine. The problem occurs when I close the `Dialog` by pressing <kbd>Esc</kbd>. Unfortunately in this case the focus won't be restored on the trigger element, it goes to the `<body>` instead.

Is it the expected behavior, or did I found a bug? [Here](https://codesandbox.io/s/snowy-water-8on9u) you can find an example that reproduces the above-mentioned behavior.

Looking forward to your answer.

**Comment by okonet** [comment]
Same issue is reported in Reakit: https://github.com/reakit/reakit/issues/855 but there is no resolution there too.

**Comment by snowystinger** [comment]
It seems like a bug from a quick look. I can reproduce it. Would you mind opening an Issue and linking it to this?
